agent/xds/listeners.go

```diff
@@ -1541,7 +1541,7 @@ func makeCommonTLSContextFromLeaf(cfgSnap *proxycfg.ConfigSnapshot, leaf *struct
 	}
```

Comment:
You could slightly simplify the plumbing here if you added:
```
if tlsParams == nil {
    tlsParams = &envoy_tls_v3.TlsParameters{}
}
```
rather than forcing the caller to create an empty object in most cases as you have done.

Reply:
Added a commit doing this, but the UX only feels a little cleaner since AFAIK it's not possible to overload a function with multiple signatures or optional parameters in Go so I added `makeCommonTLSContextFromLeafWithoutParams` as a wrapper function passing `nil`